src/main/scala/io/iohk/ethereum/domain/Address.scala

```diff
@@ -33,6 +36,12 @@ object Address {
     require(bytes.length <= Length, s"Invalid address: $hexString")
     Address(bytes)
   }
+
+  def apply(keyPair: AsymmetricCipherKeyPair): Address = {
+    //byte 0 of encoded ECC point indicates that it is uncompressed point, it is part of bouncycastle encoding
+    val pub = keyPair.getPublic.asInstanceOf[ECPublicKeyParameters].getQ.getEncoded(false).tail
```

Comment:
Couldn't we use `crypto.pubKeyFromKeyPair`?

Reply:
Good point!